core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -524,10 +669,37 @@ private[spark] class MapOutputTrackerMaster(
     }
   }
 
+  def registerMergeResult(shuffleId: Int, reduceId: Int, status: MergeStatus) {
+    shuffleStatuses(shuffleId).addMergeResult(reduceId, status)
+  }
+
+  def registerMergeResults(shuffleId: Int, statuses: Seq[(Int, MergeStatus)]): Unit = {
+    statuses.foreach {
+      case (reduceId, status) => registerMergeResult(shuffleId, reduceId, status)
+    }
+  }
+
+  def unregisterMergeResult(shuffleId: Int, reduceId: Int, bmAddress: BlockManagerId) {
```

Comment:
This's only called in tests yet. Do you plan to use it in the following PR? Although, I think we can also complete it in this PR.

Reply:
Are you suggesting combining SPARK-32921 for MapOutputTracker changes with SPARK-32920 for DAGScheduler changes in this PR?

Reply:
Ok. We can leave it to the following PR since you already have a planned ticket for it.

Reply:
Want to clarify further here.
Even with SPARK-32920, we are not currently using this API.
This is because we would fallback to fetching the unmerged original shuffle blocks if encountering any fetch failures on merged shuffle files.
This would hide all such fetch failures from the DAGScheduler, thus not invoking this unregisterMergeResult API.
This is a potential area for future improvement, however it won't lead to any potential data duplication issues.
Note that, when we finalize a shuffle merge, where the driver has already received the MergeStatus for a given shuffle, we won't enable push for following retries of the shuffle map stage sharing the same shuffle dependency (#30164).
This will make sure that the MergeStatus always track the correct information about the merged shuffle partitions.

Reply:
Thanks for clarify. But I think we usually don't add functions when they're not necessary. We can always add them back when we needed in future improvement. For now, I think they only introduce unnecessary review work. So I'd suggest to remove it first.

Reply:
This method in our most recent internal version is being used by DAGScheduler to unregister merge result in certain cases.
It will be reflected with SPARK-32920.
Would like to still keep it in this PR.

Reply:
@venkata91 The signature of this method is outdated. In the current version it also accepts an optional mapId. Can you please pull that change?

Reply:
Changed to the recent signature we have internally.

Reply:
@venkata91 We also added a UT to test the method when optional map id is provide. Please pull that in as well.

Reply:
Done